backend/modules/dataloaders/truefoundryloader.py

```diff
@@ -23,98 +24,100 @@ def load_filtered_data(
         data_ingestion_mode: DataIngestionMode,
     ) -> Iterator[List[LoadedDataPoint]]:
         """
-        Loads data from a truefoundry data directory / artifact with FQN specified by the given source URI.
+        Loads data from a truefoundry data directory with FQN specified by the given source URI.
         """
-        truefoundry_client = ml.get_client()
-        artifact_files_dir = None
-        datasource_type = None
-
-        if data_source.uri.startswith("artifact"):
-            datasource_type = "artifact"
-            # uri should be the artifcat fqn that includes the version number
-            artifact_fqn = data_source.uri
-            # Get information about the data directory and download it to the destination directory.
-            logger.info("Downloading Artifact: {}".format(artifact_fqn))
-
-            # Get the artifact version directly
-            dataset = truefoundry_client.get_artifact_version_by_fqn(artifact_fqn)
-            # download it to disk
-            # `artifact_files_dir` points to a directory that has all contents of the artifact
-            artifact_files_dir = dataset.download(path=dest_dir)
-            logger.debug(f"Artifact download info: {artifact_files_dir}")
-
-        elif data_source.uri.startswith("data-dir"):
-            datasource_type = "data-dir"
-            # Data source URI is the FQN of the data directory.
-            dataset = truefoundry_client.get_data_directory_by_fqn(data_source.uri)
-            artifact_files_dir = dataset.download(path=dest_dir)
-            logger.debug(f"Data directory download info: {artifact_files_dir}")
-
-        else:
-            raise ValueError(f"Unsupported data_source uri type {data_source.uri}")
-
-        if artifact_files_dir:
-            if os.path.exists(os.path.join(artifact_files_dir, "files")):
-                logger.debug("Files directory exists")
-                artifact_files_dir = os.path.join(artifact_files_dir, "files")
+        tfy_files_dir = None
+
+        # Check if the data source type is allowed
+        if not any(
+            data_source.type.startswith(allowed_type)
+            for allowed_type in ALLOWED_DATA_SOURCE_TYPES
+        ):
+            raise ValueError(f"Unsupported data source type {data_source.type}")
+
+        # If the data source type is data-dir, download the data directory from TrueFoundry.
+        if data_source.uri.startswith("data-dir"):
```

Comment:
We can remove this if for now, otherwise, add a corresponding else.

Reply:
I'll remove this check since we will only allow data-dir

Reply:
Removed in latest commits